sklearn/utils/extmath.py

```diff
@@ -839,18 +748,28 @@ def _incremental_mean_and_var(X, last_mean, last_variance, last_sample_count):
     # new = the current increment
     # updated = the aggregated stats
     last_sum = last_mean * last_sample_count
-    new_sum = _safe_accumulator_op(np.nansum, X, axis=0)
+    if sample_weight is not None:
+        new_sum = _safe_accumulator_op(np.nansum, X * sample_weight[:, None],
+                                       axis=0)
+        new_sample_count = np.sum(sample_weight[:, None] * (~np.isnan(X)),
+                                  axis=0)
+    else:
+        new_sum = _safe_accumulator_op(np.nansum, X, axis=0)
+        new_sample_count = np.sum(~np.isnan(X), axis=0)
```

Comment:
If we are really digging for performance, ``np.count_nonzero(~np.isnan(X), axis=0)`` can be much faster

Reply:
Slightly better:
```
without above change
%timeit scale_with_partial_fit(X, sample_weight=None)
1.13 s ± 75.5 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

with above change
%timeit scale_with_partial_fit(X, sample_weight=None)
1.08 s ± 4.41 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

Reply:
Should we go with that?

Reply:
@jeremiedbb Could you add this as suggested change so that the authorship is properly recorded. Thanks.

Reply:
Hum it's weird, when I first compared the timings I got a huge speed-up but when I try again I can't reproduce. Let's leave it as is and merge :)

Reply:
Fine with me, thanks for checking again